### ISSUE TITLE
MultiLabel SVM: not cancellable

### ISSUE THREAD

**Issue by mhusinsky** [accord-net/framework#182]
Hello Cesar,

i just noticed that unlike the KSVM and MulticlassSVM the MultilableSVM has no Run method that takes a CancellationToken as an argument, which probably means that it cannot be executed in a task and be cancelled cooperatively.

I'd like to post this as a feature request to have similar features in all SVM training classes.
Thanks for your work!

**Comment by cesarsouza** [comment]
Hi Mathias,

Thanks, I will add support for this! However, could you please let me know how you are using this feature? As you might have seen, the SVMs are the only models that actually support collaborative cancellation. 

I am in the process of refactoring some parts of the framework after this merge with AForge.NET, and I plan to bring more consistency to the framework by enforcing some common mechanisms for learning, cancelling, etc. If you have any suggestions for improving it (that would possibly require breaking changes), please let me know as this would be the right time for implementing them!

Regards,
Cesar

**Comment by mhusinsky** [comment]
Hi Cesar,

I stumbled over this while updating a set of plugins for the visual programming environment *VVVV* (http://vvvv.org , https://vvvv.org/contribution/vvvv.packs.machinelearning-%28alpha%29) that are based on your framework and I programmed a few months ago. Since VVVV is an "always on" runtime which is great for many forms of media programming and prototyping, especially when it comes to using live input from devices like the Kinect, LeapMotion, ... but also Video and Audio and other signals. It is also great for many kinds of realtime DataViz using DirectX as a powerful graphics engine and much more.

Having the possibility to perform machine learning tasks in an environment like this was my big wish and accord proved to be a great choice as VVVV has a great integration of the .NET world via its plugin interface.
One big issue is still the single threaded runtime nature of VVVV which is why I'd like to make all computing intensive tasks (like running a learning algorithm) asynchronous. Not doing so stalls the environment which might become problemativ when online learning tasks are to be done.

I'm quite new to the concept of tasks/threads but was glad to find out that the SVMs support this already. The plugins for VVVV using the SVMs are implemented using Generics which helped me to reduce redundancy in the code a lot. Since MultilabelSVMs don't support cancellation (for now) I had to introduce some workarounds and training is preformed synchronously. Thanks for adding support for collaborative cancellation because that way the wrapping into plugins can be done in a much cleaner way.

I don't have any precise suggestions on how to streamline framework right now (except for making computationally intensive tasks run asynchronously). For sure it would be a good thing to have similar interfaces and behaviour on different learning algorithms (i.e. how to configure and run/cancel them) but I am aware that this might not be possible because of their nature. Anyway, I will keep you updated with suggestions as soon as something comes up :)

Thanks again and all the best,
Matthias
